agent/src/main/java/org/openremote/agent/protocol/AbstractProtocol.java

```diff
@@ -510,7 +511,7 @@ public String toString() {
     /**
      * Link an attribute to its linked protocol configuration.
      */
-    abstract protected void doLinkAttribute(AssetAttribute attribute, AssetAttribute protocolConfiguration);
+    abstract protected void doLinkAttribute(AssetAttribute attribute, AssetAttribute protocolConfiguration) throws IOException;
```

Comment:
Please change to `Exception` not `IOException`, attributes shouldn't be establishing the connection that is the job of the `doLinkProtocolConfiguration` so shouldn't see `IOException` here in general so `throws` shouldn't be limited  to that type